planning/behavior_velocity_planner/src/scene_module/crosswalk/scene_walkway.cpp

```diff
@@ -37,72 +43,127 @@ WalkwayModule::WalkwayModule(
   planner_param_ = planner_param;
 }
 
-bool WalkwayModule::modifyPathVelocity(
-  PathWithLaneId * path, tier4_planning_msgs::msg::StopReason * stop_reason)
+boost::optional<std::pair<double, geometry_msgs::msg::Point>> WalkwayModule::getStopLine(
+  const PathWithLaneId & ego_path) const
 {
+  const auto & ego_pos = planner_data_->current_pose.pose.position;
+
+  const auto stop_line = getStopLineFromMap(module_id_, planner_data_, "crosswalk_id");
+  if (stop_line) {
+    const auto intersects = getLinestringIntersects(
+      ego_path, lanelet::utils::to2D(stop_line.get()).basicLineString(), ego_pos, 2);
+    if (!intersects.empty()) {
+      const auto p_stop_line =
+        createPoint(intersects.front().x(), intersects.front().y(), ego_pos.z);
+      const auto dist_ego_to_stop = calcSignedArcLength(ego_path.points, ego_pos, p_stop_line);
+      return std::make_pair(dist_ego_to_stop, p_stop_line);
+    }
+  }
+
+  {
+    if (!path_intersects_.empty()) {
+      const auto p_stop_line = path_intersects_.front();
+      const auto dist_ego_to_stop = calcSignedArcLength(ego_path.points, ego_pos, p_stop_line) -
+                                    planner_param_.stop_line_distance;
+      return std::make_pair(dist_ego_to_stop, p_stop_line);
+    }
+  }
+
+  return {};
+}
+
+bool WalkwayModule::modifyPathVelocity(PathWithLaneId * path, StopReason * stop_reason)
+{
+  const auto & base_link2front = planner_data_->vehicle_info_.max_longitudinal_offset_m;
+
   debug_data_ = DebugData();
-  debug_data_.base_link2front = planner_data_->vehicle_info_.max_longitudinal_offset_m;
-  first_stop_path_point_index_ = static_cast<int>(path->points.size()) - 1;
```

Comment:
@satoshi-ota 
This is refactoring PR but do you want to remove this line?

Reply:
I use motion_utils::insertTargetPoint() in walkway module and crosswalk module, so `first_stop_path_point_index_` is unused variable